applications/FluidDynamicsApplication/custom_utilities/fluid_auxiliary_utilities.cpp

```diff
@@ -343,6 +373,113 @@ FluidAuxiliaryUtilities::ModifiedShapeFunctionsFactoryType FluidAuxiliaryUtiliti
     }
 }
 
+void FluidAuxiliaryUtilities::InternalCreateSlipMultiPointConstraints(
+    ModelPart& rModelPart,
+    const Variable<array_1d<double,3>>& rUnknownVar,
+    const Flags& rSlipFlag,
+    const std::function<double(const NodeType&, const array_1d<double,3>&)>& rWallNormalVelocityCalculator)
```

Comment:
this looks good to me, however speak about this with   @EduardGomezEscandell since he has done some experimentation about performance implications

Reply:
In my benchmark I was comparing the speed of reassigning the function pointer / std::function between two different functions many times, and calling it a few extra times. In this case there the raw pointer was faster.

If you're assigning it once and calling it many times then I don't know which option is best.